lib/Elastica/Document.php

```diff
@@ -36,16 +37,14 @@ class Document extends AbstractUpdateAction
     /**
      * Creates a new document.
      *
-     * @param int|string   $id    OPTIONAL $id Id is create if empty
+     * @param string       $id    OPTIONAL $id Id is create if empty
      * @param array|string $data  OPTIONAL Data array
-     * @param Type|string  $type  OPTIONAL Type name
      * @param Index|string $index OPTIONAL Index name
      */
-    public function __construct($id = '', $data = [], $type = '', $index = '')
+    public function __construct(string $id = null, $data = [], $index = '')
```

Comment:
string and `null`?

Reply:
True, as the id will be generated if not given